### ISSUE TITLE
Improve usability of house address input (number and road)

### ISSUE THREAD

**Issue by sjvudp** [streetcomplete/StreetComplete#5967]
**Use case**

When adding the addresses of many houses, typically walking along some road the current procedure is like this:

1. Touch the house number icon: A Popup appears with the last number entered. You enter the number and confirm/close
2. The street name icon appears, you touch it: A popup appears asking for the street name, and the map zooms in (why, BTW?). You typically zoom out again to touch the correct road (in most cases; otherwise you enter the name). You confirm/close the dialog

**Proposed Solution**

My initial idea was:

- As the dialog for the house has increment/decrement buttons, shouldn't the street name of the last number be kept, saving the complete second step when using the increment/decrement buttons (instead of entering a new number)?

However thinking about it, I wonder:

- Shouldn't both inputs (number and road) be unified to one dialog, like this?:
  - House number (last) with increment/decrement button (as it it now)
  - Road name (displaying last name) with a pre-selected check-box "same road"

Only if you uncheck the "same road" checkbox, a field to input the new road name should appear.

Obviously this saves a lot of interactions, and maybe even reduces the amount of changes to upload.

I think that new dialog could also be used to edit existing addresses; maybe then the "same road" checkbox should be renamed to "keep road name" (or similar).

**Comment by westnordost** [comment]
It does save some interactions, but the primary design goal of StreetComplete is not to be as efficient as possible but to enable contributing on the side, i.e. have atomic quests - always ask for the smallest possible chunk of information.

You can always add more knobs and switches, but then at some point, the app looks like Vespucci.

In any case, if you are out for efficiency, there *is* actually a way with StreetComplete to do this. Use the address layer. In the address layer, you can specify house number and street name at the same time.

That being said, I will keep this in mind. Maaaybe I will reconsider and put both quests into one quest, but only during/after migrating the UI to Compose.

**Comment by mnalis** [comment]
**TL;DR:** while I understand (and personally support!) the wish to make the workflow faster, perhaps it is better **_not_** to change how address quest work - there seems to be several layers of hidden problems, and our eventual "solution" might turn out not to be good  (or consistent) enough.

Or, alternatively, there should be discussion on potential solution _before_ it is implemented, so people might be able to spot problems with it.

---

> That being said, I will keep this in mind. Maaaybe I will reconsider and put both quests into one quest, but only during/after migrating the UI to Compose.

Hm, I'm not sure if that would be a good idea to merge those two quests into one...

For example, it is very common here that housenumber **plates do _NOT_ show street name** at all, but only housenumber:

![Image](https://github.com/user-attachments/assets/89ad1a81-125e-402b-bfbb-d513a469fc46)

Or, it is often the case that the housenumber is in much bigger font size so is legible (even for houses with gardens in front so further away) while the street name is completely unreadable. E.g. something like this `addr:housenumber=41`:

![Image](https://github.com/user-attachments/assets/889cfad0-b7f6-4cf0-beab-ab1b9f5fbfe1)


So often one can only input house number, not a street name. That is where _"⚛️ Atomic quests"_ comes quite handy. You solve one quest, and hide the other.

Now, it perhaps might be worked around by allowing one to specify **only** housenumber without specifying street in that newly proposed combined _"housenumber & street name quest"_. Because, if that was **not** allowed, it would violate _"🤔 No unanswerable quests"_ too, and very likely lead users to guessing (sometimes wrongly), which would be very bad outcome (much worse then slightly less efficient address collecting).

But, even if that workaround is implemented (which is a "must", IMHO), that brings extra problems:
- if there is no other quest, there perhaps should be a warning on a combined asking user to confirm if they want to only enter housenumber without street name[^1] to prevent unintended half-answers

- also, the other quest  just for "Street name" should remain anyway, for those [dozens of millions of cases](https://taginfo.openstreetmap.org/keys/addr:housenumber#combinations) when housenumber is present, but street name isn't [^2] . 

So we would end up with two quests anyway: one quest which is _"Combined housenumber & street name"_, and another quest which is _"Street name only"_ (i.e. same as existing one), which would be somewhat inconsistent and confusing[^3]. 

I _think_ that implementing all that might work from technical side (for my use cases anyway), but I'm not sure it would be so good from UX side (for the average newbie user anyway) 🤷‍♂️ . 

[^1]: however, such warning should be dismissable for the session (akin to _"are you sure you visited this place"_ popups), in case user is in the area where it is common for houses to only have housenumber displayed.
[^2]:  e.g. when StreetComplete mapper with a better eyesight (for second picture example above comes), or where the street name is not entered for any other reason (i.e. `addr:housenumber` was not entered by SC mappers)
[^3]: e.g. you solve the first combined quest and answer that there is only housenumber (and no street name!) specified, and then the quest asking you to "specify street name" pops up. There could be workarounds that add `signed:addr:street=no` (or similar) to prevent that, but that just makes the "solution" even more ugly and complex.
